relayer/crates/starknet-chain-components/src/types/wallet.rs

```diff
@@ -3,9 +3,37 @@ use starknet::core::types::Felt;
 
 use crate::impls::types::address::StarknetAddress;
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum SigningKey {
+    File { file: String },
+    Felt { felt: Felt },
+}
+
+impl TryFrom<SigningKey> for Felt {
+    type Error = String;
+
+    fn try_from(value: SigningKey) -> Result<Self, Self::Error> {
+        match value {
+            SigningKey::File { file } => Ok(std::fs::read_to_string(file)
+                .map_err(|e| format!("{e}"))?
+                .trim()
+                .parse()
+                .map_err(|e| format!("{e}"))?),
+            SigningKey::Felt { felt } => Ok(felt),
+        }
+    }
+}
+
+impl From<Felt> for SigningKey {
+    fn from(felt: Felt) -> Self {
+        Self::Felt { felt }
+    }
+}
+
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct StarknetWallet {
     pub account_address: StarknetAddress,
-    pub signing_key: Felt,
+    pub signing_key: SigningKey,
```

Comment:
I think the signing key here should always be `Felt`. We should update the config to load the key from a path, and load the key before initializing the chain or wallet.

Reply:
Can you please take over this PR? Maybe I have to do the same way you do it for `CosmosChainConfig`

https://github.com/informalsystems/hermes-sdk/blob/27168d1ee8c004d87b39d769e80c686d79ed53c5/crates/cosmos/cosmos-chain-components/src/impls/types/config.rs#L21-L22

Reply:
5884df0